### PR TITLE
Some refactor and clean up with KeyBinding and EntryListWidget$Entry

### PR DESCRIPTION
**Pull request by shedaniel**
Signed-off-by: shedaniel <daniel@shedaniel.me>

### CODE REVIEW

File: mappings/net/minecraft/client/gui/screen/options/ControlsListWidget.mapping [hunk 1]
```diff
@@ -5,11 +5,15 @@ CLASS net/minecraft/class_459 net/minecraft/client/gui/screen/options/ControlsLi
 		ARG 1 gui
 		ARG 2 client
 	CLASS class_460 CategoryEntry
-		FIELD field_2736 name Lnet/minecraft/class_2561;
-		FIELD field_2737 nameWidth I
+		FIELD field_2736 categoryText Lnet/minecraft/class_2561;
+		FIELD field_2737 categoryTextWidth I
+		METHOD <init> (Lnet/minecraft/class_459;Lnet/minecraft/class_2561;)V
+			ARG 2 categoryText
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/options/ControlsListWidget.mapping]:
```suggestion
		FIELD field_2736 text Lnet/minecraft/class_2561;
		FIELD field_2737 textWidth I
		METHOD <init> (Lnet/minecraft/class_459;Lnet/minecraft/class_2561;)V
			ARG 2 text
```

File: mappings/net/minecraft/client/util/InputUtil.mapping [hunk 2]
```diff
@@ -2,38 +2,57 @@ CLASS net/minecraft/class_3675 net/minecraft/client/util/InputUtil
...
+		ARG 2 keyCallback
+		ARG 3 charModsCallback
 	METHOD method_15987 isKeyPressed (JI)Z
 		ARG 0 handle
+		ARG 2 code
 	METHOD method_21735 isRawMouseMotionSupported ()Z
 	METHOD method_21736 setRawMouseMotionMode (JZ)V
 		ARG 0 window
 		ARG 2 value
 	CLASS class_306 KeyCode
```
Comment (liach) [inline mappings/net/minecraft/client/util/InputUtil.mapping]: maybe just `Key` for this class so `fromKeyCode` returning a `Key` makes sense (than returning a `KeyCode`)

File: mappings/net/minecraft/client/util/InputUtil.mapping [hunk 3]
```diff
@@ -2,38 +2,57 @@ CLASS net/minecraft/class_3675 net/minecraft/client/util/InputUtil
...
+			ARG 1 other
+		METHOD method_1441 getTranslationKey ()Ljava/lang/String;
 		METHOD method_1442 getCategory ()Lnet/minecraft/class_3675$class_307;
-		METHOD method_1444 getKeyCode ()I
+		METHOD method_1444 getCode ()I
+		METHOD method_27445 getLocalizedText ()Lnet/minecraft/class_2561;
 	CLASS class_307 Type
 		FIELD field_1673 name Ljava/lang/String;
 		FIELD field_1674 map Lit/unimi/dsi/fastutil/ints/Int2ObjectMap;
+		FIELD field_24197 textLocalizeFunction Ljava/util/function/BiFunction;
```
Comment (liach) [inline mappings/net/minecraft/client/util/InputUtil.mapping]:
```suggestion
		FIELD field_24197 textTranslator Ljava/util/function/BiFunction;
```
don't need the `function` suffix

**Review by liach — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/client/options/KeyBinding.mapping [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
 	FIELD field_1652 keyCategories Ljava/util/Set;
 	FIELD field_1653 pressed Z
-	FIELD field_1654 defaultKeyCode Lnet/minecraft/class_3675$class_306;
-	FIELD field_1655 keyCode Lnet/minecraft/class_3675$class_306;
+	FIELD field_1654 defaultKey Lnet/minecraft/class_3675$class_306;
+	FIELD field_1655 boundKey Lnet/minecraft/class_3675$class_306;
 	FIELD field_1656 categoryOrderMap Ljava/util/Map;
 	FIELD field_1657 keysById Ljava/util/Map;
-	FIELD field_1658 keysByCode Ljava/util/Map;
+	FIELD field_1658 keyBindingsByKey Ljava/util/Map;
```
Comment (liach) [inline mappings/net/minecraft/client/options/KeyBinding.mapping]: `keyToBindings`?

Comment (shedaniel) [reply mappings/net/minecraft/client/options/KeyBinding.mapping]: that works

**Comment by shedaniel** [comment]
updated to 20w20a

File: mappings/net/minecraft/client/util/InputUtil.mapping [hunk 5]
```diff
@@ -1,39 +1,58 @@
...
-		METHOD method_1444 getKeyCode ()I
+		METHOD method_1444 getCode ()I
+		METHOD method_27445 getLocalizedText ()Lnet/minecraft/class_2561;
 	CLASS class_307 Type
 		FIELD field_1673 name Ljava/lang/String;
 		FIELD field_1674 map Lit/unimi/dsi/fastutil/ints/Int2ObjectMap;
+		FIELD field_24197 textTranslator Ljava/util/function/BiFunction;
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;Ljava/util/function/BiFunction;)V
+			ARG 3 name
+			ARG 4 textLocalizeFunction
```
Comment (liach) [inline mappings/net/minecraft/client/util/InputUtil.mapping]:
```suggestion
			ARG 4 textTranslator
```

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/util/InputUtil.mapping (hunk 5 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/util/InputUtil.mapping]: Missed this oops

Comment (liach) [reply mappings/net/minecraft/client/util/InputUtil.mapping]: Yeah I should've suggested this change when I suggested the field change. Sorry!

File: mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping [hunk 6]
```diff
@@ -95,12 +95,14 @@ CLASS net/minecraft/class_350 net/minecraft/client/gui/widget/EntryListWidget
...
 			ARG 1 matrices
-			ARG 2 x
-			ARG 3 y
-			ARG 4 width
-			ARG 5 height
-			ARG 6 mouseX
-			ARG 7 mouseY
+			ARG 2 index
+			ARG 3 x
+			ARG 4 y
```
Comment (YanisBft) [inline mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping]:
```suggestion
			ARG 3 y
			ARG 4 x
```

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Comment by liach** [comment]
Fyi the build fails because of a network connection issue to jcenter. Mind restart that build for java 11 to fix it.

**Pull request merged**